mlflow/server/js/src/experiment-tracking/components/CompareRunContour.js

```diff
@@ -14,10 +14,10 @@ import CompareRunUtil from './CompareRunUtil';
 
 export class CompareRunContour extends Component {
   static propTypes = {
-    runInfos: PropTypes.arrayOf(RunInfo).isRequired,
-    metricLists: PropTypes.arrayOf(Array).isRequired,
-    paramLists: PropTypes.arrayOf(Array).isRequired,
-    runDisplayNames: PropTypes.arrayOf(String).isRequired,
+    runInfos: PropTypes.arrayOf(PropTypes.instanceOf(RunInfo)).isRequired,
+    metricLists: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.object)).isRequired,
+    paramLists: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.object)).isRequired,
```

Comment:
We can be more strict here using `PropTypes.shape`.

Reply:
> Hopefully going forward we can spend some time converting to typescript

Agree with using typescript which should make the UI development much easier.

> Hopefully going forward we can spend some time converting to typescript。

I'm actually working on this. I'll submit a PR when it's ready.